examples/client/lwm2mclient.c

```diff
@@ -118,14 +126,30 @@ static void prv_quit(lwm2m_context_t * lwm2mH,
     (void)buffer;
     (void)user_data;
 
+    // Set the quit flag to terminate the program
     g_quit = 1;
 }
 
+/**
+ * @brief Signal handler for SIGINT (Ctrl+C) for graceful shutdown.
+ *
+ * @param lwm2mH LwM2M context.
+ * @param uri URI of the resource.
+ */
 void handle_sigint(int signum)
 {
+    // Set the quit flag to initiate a graceful shutdown
     g_quit = 2;
```

Comment:
This flag should be also described in a comment in the start of the file, where it is defined.